### PR TITLE
Fix/query lighthouse

### PR DESCRIPTION
**Pull request by tbushman**
@Bouncey I don't know what I did wrong, but I was not able to push back to the other branch. :/ Very sorry.  
I added tests for teams. Thanks.

### CODE REVIEW

File: server/team/team.contoller.js [hunk 1]
```diff
@@ -2,20 +2,109 @@ const debug = require('debug');
...
+  .then(() => {
+    if (!newTeam) {
+      return res.status(500).json({ acknowladged: false });
+    }
+    return User.findOneAndUpdate(
+      { _id: req.user._id },
+      { $set: { teamId: newTeam._id } }
+    )
+    .then((user) => {
+      res.redirect(`/api/teams/${user.teamId}`);
```
Comment (tbushman) [inline server/team/team.contoller.js]: Sorry, this should probably be res.redirect('/team') ... and same with update()

File: server/team/team.contoller.js [hunk 2]
```diff
@@ -2,20 +2,109 @@ const debug = require('debug');
...
+  .then((team) => {
+    if (!team) {
+      return res.redirect('/team');
+    }
+    return res.status(200).json(team);
+  });
+}
+
+function deleteTeam(req, res) {
+  return Team.remove({ _id: req.params._id })
```
Comment (tbushman) [inline server/team/team.contoller.js]: this should be req.params.teamId

File: server/team/team.contoller.js [hunk 3]
```diff
@@ -2,20 +2,109 @@ const debug = require('debug');
...
+        });
+      });
+    }
+    return updateTeamScore();
+  });
+}
+
+function list(req, res) {
+  return Team.find({})
+  .sort()
```
Comment (tbushman) [inline server/team/team.contoller.js:85]: eventually sort by rank

File: config/express.js [hunk 4]
```diff
@@ -21,6 +22,13 @@ const app = express();
 if (config.env === 'development') {
   app.use(logger('dev'));
 }
+const store = new MongoDBStore({
+  mongooseConnection: mongoose.connection,
+  collection: 'mySessions'
+});
+store.on('error', (error) => {
+  logger(error);
+});
```
Comment (tbushman) [inline config/express.js:31]: no more errors, but the session isn't available in the template ... yet

**Comment by tbushman** [comment]
@Bouncey I believe this addresses all of the review comments in #8 . The form body is working with multer but session in template will go in a separate PR since this pr is already too full. Thanks for your review.

File: server/team/team.contoller.js [hunk 5]
```diff
@@ -2,20 +2,111 @@ const debug = require('debug');
...
+  .lean()
+  .then((teams) => {
+    if (teams.length === 0) {
+      return res.render('createTeam');
```
Comment (Bouncey) [inline server/team/team.contoller.js:89]: This check could potentially throw an error if the query returns `null`.

This is an API call, we shouldn't be rendering here. If teams is null or empty, return an empty array.

File: server/team/team.contoller.js [hunk 6]
```diff
@@ -2,20 +2,111 @@ const debug = require('debug');
...
+    if (!team) {
+      return res.redirect('/team');
+    }
+    return res.status(200).json(team);
+  });
+}
+
+function deleteTeam(req, res) {
+  return Team.remove({ _id: req.params.teamId })
+  .then(team => res.status(200).json(`deleted team ${team.name}`));
```
Comment (Bouncey) [inline server/team/team.contoller.js:109]: We should have a `.catch()` here that passes the error to `next()`. That way we can handle the error down stream with something like rollbar

File: server/team/team.test.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+const chai = require('chai'); // eslint-disable-line import/newline-after-import
+const expect = chai.expect;
+const app = require('../../index');
+
+chai.config.includeStack = true;
+
+/**
+ * root level hooks
+ */
+after((done) => {
```
Comment (Bouncey) [inline server/team/team.test.js:13]: We can mock methods on a mongoose model so we do not encounter these types of gotchas

File: server/views/partials/nav.pug [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+      i.fas.fa-bars
+    #navbarResponsive.collapse.navbar-collapse
+      ul.navbar-nav.ml-auto
+        if (!session || !session.userId)
+        else
+          li.nav-item.mx-0.mx-lg-1
+            a.nav-link.py-3.px-0.px-lg-3.rounded(href=`/team`) Team
+        li.nav-item.mx-0.mx-lg-1
+          if (!session || !session.userId)
+            a.nav-link.py-3.px-0.px-lg-3.rounded(href= `/api/auth/github/callback`) Signin
```
Comment (Bouncey) [inline server/views/partials/nav.pug:16]: Ok, I see what you did there... At first I thought this shouldn't work and that there was an issue with it. But. now I realise I just had an extra step in the mix.

File: server/views/partials/nav.pug [hunk 9]
```diff
@@ -0,0 +1,23 @@
+// Navigation
+nav#mainNav.navbar.navbar-expand-lg.bg-secondary.fixed-top.text-uppercase
+  .container
+    a.navbar-brand.js-scroll-trigger(href='#page-top') Start Bootstrap
+    button.navbar-toggler.navbar-toggler-right.text-uppercase.bg-primary.text-white.rounded(type='button', data-toggle='collapse', data-target='#navbarResponsive', aria-controls='navbarResponsive', aria-expanded='false', aria-label='Toggle navigation')
+      | Menu
+      i.fas.fa-bars
+    #navbarResponsive.collapse.navbar-collapse
+      ul.navbar-nav.ml-auto
+        if (!session || !session.userId)
```
Comment (Bouncey) [inline server/views/partials/nav.pug:10]: Maybe we could just use something related to `req.user`? `req.user` only exists if passport has authenticated them.

File: server/team/team.contoller.js (hunk 5 above)
Comment (Bouncey) [reply server/team/team.contoller.js:89]: See this screenshot
image

**Review by Bouncey — APPROVED** [review]
Although this. PR does require changes, I am going to merge it in an effort to keep moving. It will be used tomorrow!!!

**Comment by Bouncey** [comment]
@tbushman Can you please keep working on the review comments. I am merging so our branches do not diverge too far

**Pull request merged**